### PR TITLE
Job post crud

### PR DESCRIPTION
**Pull request by jjhiggz**
I was going to do CRUD, but instead I decided to work on filtering. I think there needed to be a strategy for users to see their own job postings before approval. That way you can actually test out the crud stuff and make sure that it works.

In order to get there I wanted to debug some of the filtering stuff. I created a "useFilteredState" hook and wrote tests for it. Now the filtering actually works. I think next add like to add some UI to indicate whether or not the job posting has been approved. I'm not sure how that approval process actually works or how it ties into crud though, would love to have a discussion on it.

### CODE REVIEW

**Comment by danhannigan** [comment]
Looks like I get an error when toggling the new filter toggles, possibly related to me being logged out?

Kapture 2022-03-03 at 11 19 07

Might be worth hiding those toggles when there is no authenticated user, too.

File: pages/jobs/index.jsx [hunk 1]
```diff
@@ -87,23 +142,49 @@ export default function BrowseJobsPage({ hiringEntries, lookingEntries }) {
                 <Heading as="h5" size="sm" mb="4">
                   Filters
                 </Heading>
-                <Switch onChange={() => toggleRemoteOnly(!remoteOnly)}>Remote Only</Switch>
+                <Switch
```
Comment (danhannigan) [inline pages/jobs/index.jsx:145]: Let's hide these if there's no active user.

File: utils/filters/job.filters.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  filterName: "isRemote",
+  filterFn: (item) => item.location.includes("remote"),
+};
+
+export const createIsMineFilter = (user) => ({
+  filterName: "isMine",
+  filterFn: (item) => item.user_id === user.id,
+});
+
+export const createIsNotMineFilter = (user) => ({
```
Comment (danhannigan) [inline utils/filters/job.filters.js:11]: As I'm thinking through some interactions with this, I think we can scrap it and just stick with the "isMine" toggle.

**Review by danhannigan — CHANGES_REQUESTED** [review]
Check out the bit of feedback I have, and see if you can replicate that error when toggling those filters while logged out.

File: utils/filters/job.filters.js (hunk 2 above)
Comment (jjhiggz) [reply utils/filters/job.filters.js:11]: good call, on it

**Review by danhannigan — APPROVED** [review]

**Pull request merged**